src/main/java/net/sf/mzmine/modules/peaklistmethods/peakpicking/deconvolution/DeconvolutionTask.java

```diff
@@ -59,6 +61,11 @@
     // User parameters
     private final ParameterSet parameters;
 
+	private RSessionWrapper rSession;

```

Comment:
If I am not mistaken, CentWave is a singleton itself, instantiated in DeconvolutionParameters:
```
    private static final PeakResolver[] RESOLVERS = {
	    new BaselinePeakDetector(), new NoiseAmplitudePeakDetector(),
	    new SavitzkyGolayPeakDetector(), new MinimumSearchPeakDetector(),
	    new CentWaveDetector() };
```

So although you are right that resolvePeaks() is called many times, the R-related properties could be initialized in the CentWaveDetector's constructor... or not?